web/src/ui/components/RequestBuilder.js

```diff
@@ -0,0 +1,114 @@
+import { Select, JsonTextarea } from "../commons"
+import { isValidJson, getStringifiedJSON } from "../../util"
+import { useState } from "react"
+
+/*
+	TODO
+	JSON Textarea formatierungen
+	Path Logik auf eigene Komponente
+
+*/
+
+const RequestBuilder = () => {
+    const [method, setMethod] = useState("")
+    const [path, setPath] = useState("")
+    const [header, setHeader] = useState({})
+    const [body, setBody] = useState({})
+    const [bodyDisabled, setBodyDisabled] = useState(true)
+    var headerInputTimeout, bodyInputTimeout
+
+    const handleMethodChange = (selectedMethod) => {
+        setMethod(selectedMethod)
+        enableOptions(selectedMethod)
+    }
+
+    const removeCurrentOptions = () => {
+        setBodyDisabled(true)
+        //...
+    }
+
+    const enableOptions = (selectedMethod) => {
+        removeCurrentOptions()
+        if (selectedMethod.toLowerCase() === "post") {
+            setBodyDisabled(false)
+        }
+        //...
+    }
+
+    const handlePathChange = (event) => {
+        setPath(event.target.value)
+    }
+
+    const handleHeaderChange = (header) => {
+        clearTimeout(headerInputTimeout)
+        headerInputTimeout = setTimeout(() => {
+            if ((header = isValidJson(header))) {
+                setHeader(header)
+            } else {
+                //evtl Error bars triggern etc
+            }
+        }, 2000)
+    }
+    const handleBodyChange = (body) => {
+        clearTimeout(bodyInputTimeout)
+        bodyInputTimeout = setTimeout(() => {
+            if ((body = isValidJson(body))) {
+                setBody(body)
```

Comment:
Mmmm...du übernimmst den body erst nach der erfolgreichen Validierung. Fände es besser aus der Textarea eine controlled-Komponente zu machen, also den value und den set reinzugeben....dann kann die Validierung hier raus und in eine eigene JsonTextArea-Komponente verschoben werden, die Komponente kann dann zum Beispiel den setter, der reingeht, noch mal kapseln und die Validierung in der Komponente triggern:
```
function JsonTextArea({ value, set, validation = true ... }) {
   const setText = !validation ? set : (value) => {
       set(value)
       triggerValidation()
   }
   ...
}

```